features/admin.roles.v2/pages/role-edit.tsx

```diff
@@ -63,32 +64,54 @@ const RoleEditPage: FunctionComponent<RoleEditPagePropsInterface> = (
         setRoleId(roleId);
     }, []);
 
+    const userRolesV3FeatureEnabled: boolean = useSelector(
+        (state: AppState) => state?.config?.ui?.features?.userRolesV3?.enabled
+    );
+    const [ roleObject, setRoleObject ] = useState<any>(undefined);
+    const {
+        data: roleObjectV2,
+        isLoading: isRoleDetailsRequestLoadingV2,
+        error: roleDetailsRequestErrorV2,
+        mutate: mutateRoleObjectV2,
+        isValidating: isRoleDetailsRequestValidatingV2
+    } = useGetRoleById(!userRolesV3FeatureEnabled ? roleId : null);
     const {
-        data: roleObject,
-        isLoading: isRoleDetailsRequestLoading,
-        error: roleDetailsRequestError,
-        mutate: mutateRoleObject,
-        isValidating: isRoleDetailsRequestValidating
-    } = useGetRoleById(roleId);
+        data: roleObjectV3,
+        isLoading: isRoleDetailsRequestLoadingV3,
+        error: roleDetailsRequestErrorV3,
+        mutate: mutateRoleObjectV3,
+        isValidating: isRoleDetailsRequestValidatingV3
+    } = useGetRoleByIdV3(userRolesV3FeatureEnabled ? roleId : null);
+
+    /**
+     * Set the role object based on the feature flag.
+     */
+    useEffect(() => {
+        if (userRolesV3FeatureEnabled) {
+            setRoleObject(roleObjectV3);
+        } else {
+            setRoleObject(roleObjectV2);
+        }
+    }, [ roleObjectV2, roleObjectV3, userRolesV3FeatureEnabled ]);
```

Comment:
We can get rid of the `roleObject` state variable, had we used `useMemo`.

```
const roleObject = useMemo( () => userRolesV3FeatureEnabled ? roleObjectV3 : roleObjectV2)
```

Reply:
441a71e8dc56be425bd66857a6c575a4a9664dd1